src/link.js

```diff
@@ -75,17 +61,69 @@ module.exports = function link(config, args) {
     asset => path.basename(asset)
   );
 
-  if (isEmpty(assets)) {
-    return;
-  }
+  const tasks = dependencies.map((dependency) => (next) => {
+    const before = (cb) => {
+      if (dependency.config.hooks && dependency.config.hooks.before) {
+        exec(dependency.config.hooks.before, function before(err, stdout, stderr) {
```

Comment:
In my opinion, we should just stick to `npm` scripts. We could either:
```json
{
   "scripts": {
       "start": "node index.js",
       "post-rnpm-link": "abc"
   } 
}
```
so it's easier for the people to follow.